### PR TITLE
Bluetooth: Mesh: Make element as rodata

### PR DESCRIPTION
**Pull request by LingaoM**
the reason is that the Mesh Profile clearly stipulates that Mesh nodes cannot change their own element definitions.

### CODE REVIEW

**Comment by LingaoM** [comment]
This resolves bullet point 2.1 from https://github.com/zephyrproject-rtos/zephyr/issues/57267:
* All bluetooth mesh elements prefix `const`

**Review by PavelVPV — DISMISSED** [review]

File: include/zephyr/bluetooth/mesh/access.h [hunk 1]
```diff
@@ -145,19 +145,23 @@ extern "C" {
...
-#define BT_MESH_ELEM(_loc, _mods, _vnd_mods)        \
-{                                                   \
-	.loc              = (_loc),                 \
-	.model_count      = ARRAY_SIZE(_mods),      \
-	.vnd_model_count  = ARRAY_SIZE(_vnd_mods),  \
-	.models           = (_mods),                \
-	.vnd_models       = (_vnd_mods),            \
+#define BT_MESH_ELEM(_loc, _mods, _vnd_mods)		\
+{							\
+	.rt		  = (void *)(void *[]) { NULL },\
```
Comment (alxelax) [inline include/zephyr/bluetooth/mesh/access.h:150]: I really do not like this magic. What is the size of unnamed structure? Why do you allocate `pointer size` memory for 2 bytes field? 
It should be like this to be precise with operated types:
```
.rt        = &(struct runtime_ctx) { .addr = 0 },\
```

File: include/zephyr/bluetooth/mesh/access.h [hunk 2]
```diff
@@ -145,19 +145,23 @@ extern "C" {
...
+	.models           = (_mods),			\
+	.vnd_models       = (_vnd_mods),		\
 }
 
 /** Abstraction that describes a Mesh Element */
 struct bt_mesh_elem {
-	/** Unicast Address. Set at runtime during provisioning. */
-	uint16_t addr;
+	/** Mesh Element runtime information */
+	struct {
```
Comment (alxelax) [inline include/zephyr/bluetooth/mesh/access.h:161]:
```suggestion
	struct runtime_ctx {
```

**Review by alxelax — CHANGES_REQUESTED** [review]

File: include/zephyr/bluetooth/mesh/access.h (hunk 2 above)
Comment (LingaoM) [reply include/zephyr/bluetooth/mesh/access.h]: Add `bt_mesh_` prefix more suitable.

File: include/zephyr/bluetooth/mesh/access.h (hunk 1 above)
Comment (PavelVPV) [reply include/zephyr/bluetooth/mesh/access.h]: The same code has been approved by you in the previous PR: https://github.com/zephyrproject-rtos/zephyr/pull/65152
Should we change this for bt_mesh_model then?

File: include/zephyr/bluetooth/mesh/access.h (hunk 2 above)
Comment (PavelVPV) [reply include/zephyr/bluetooth/mesh/access.h]: The same code has been approved by you in the previous PR: https://github.com/zephyrproject-rtos/zephyr/pull/65152
Should we change this for bt_mesh_model then?

File: include/zephyr/bluetooth/mesh/access.h (hunk 1 above)
Comment (LingaoM) [reply include/zephyr/bluetooth/mesh/access.h]: Done.

File: include/zephyr/bluetooth/mesh/access.h (hunk 2 above)
Comment (LingaoM) [reply include/zephyr/bluetooth/mesh/access.h]: Done.

**Review by PavelVPV — APPROVED** [review]

**Review by alxelax — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**